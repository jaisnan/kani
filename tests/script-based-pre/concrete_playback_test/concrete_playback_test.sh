#!/usr/bin/env bash
# Copyright Kani Contributors
# SPDX-License-Identifier: Apache-2.0 OR MIT

set +e

OUT_DIR=bin/target

echo
echo "Starting output file check..."
echo


# Check if cargo is installed
if ! command -v cargo &> /dev/null; then
    echo "cargo command not found. Please install Rust and Cargo."
    exit 1
fi

# Check if the directory containing cargo is in the PATH environment variable
if ! echo "$PATH" | grep -q "$(dirname "$(command -v cargo)")"; then
    echo "Directory containing cargo is not in PATH. Adding directory to PATH..."
    echo "export PATH=$PATH:$(dirname "$(command -v cargo)")" >> ~/.bashrc
    source ~/.bashrc
fi

echo "Running cargo test on the unit test ..."
echo

cd bin/

output=$(grep 'channel = ' ../../../../rust-toolchain.toml | cut -d '"' -f 2)
echo "$output"

# Run cargo test on the unit test
RUSTFLAGS="--cfg=kani" cargo +${output} test

cd ..

# Try to leave a clean output folder at the end
rm -rf ${OUT_DIR}

set -e
