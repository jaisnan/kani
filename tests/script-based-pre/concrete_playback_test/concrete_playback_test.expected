
running 1 test
test harnesses::kani_concrete_playback_harness_15598097466099501582 ... FAILED

failures:

---- harnesses::kani_concrete_playback_harness_15598097466099501582 stdout ----
thread 'harnesses::kani_concrete_playback_harness_15598097466099501582' panicked at 'assertion failed: !(result_1 == Ok(101) && result_2 == Err(102))', src/main.rs:12:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    harnesses::kani_concrete_playback_harness_15598097466099501582

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

error: test failed, to rerun pass `--bin bin`
